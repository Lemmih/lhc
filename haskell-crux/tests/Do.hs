module Do where


main = do
  main
  main
  v <- main
  main

