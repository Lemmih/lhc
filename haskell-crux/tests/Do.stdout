Do.main : ∀ v6. LHC.Prim.IO v6
Do.main =
  λ @.v6^1:* → 
  LHC.Prim.thenIO @.v6^1 @.v6^1
      (Do.main @.v6^1)
    (LHC.Prim.thenIO @.v6^1 @.v6^1
        (Do.main @.v6^1)
      (LHC.Prim.bindIO @.v6^1 @.v6^1
          (Do.main @.v6^1) (λ Do.v^2:v6 → 
        Do.main @.v6^1)))
